test/unit/org/apache/cassandra/index/sai/functional/CompactionTest.java

```diff
@@ -184,7 +183,7 @@ public void testCompactionWithDisabledIndexReads() throws Throwable
             verifySSTableIndexes(v2IndexName, 1);
 
             // But index queries should not be allowed.
-            assertThrows(IndexNotAvailableException.class, () -> execute("SELECT id1 FROM %s WHERE v1>=0"));
+            assertThrows(IndexNotAvailableException.class, () -> executeInternal("SELECT id1 FROM %s WHERE v1>=0"));
```

Comment:
With the new execution mode, we receive replica-side exceptions through the distributed path. So instead of the original exception with its full stack trace, we get a `ReadFailureException` without too much information. 

I'm keeping internal execution for this type of exception checks for the existing tests since I don't think it makes much difference and we can have access to the full exception.

I think this shows that the new execution path can be useful also to differentiate coordinator-side from replica-side exceptions, so we can identify cases where coordinator-side validation is needed before sending a query to die in the replicas.